File: transportador/client/pedidos-consumer.js
const client = require('./pulsar-client').client;
const service  = require('../services/pedidos');

(async () => {
 
  const consumer = await client.subscribe({
    topic: 'persistent://public/default/pedidos',
    subscription: 'sub1',
    subscriptionType: 'Shared',
    ackTimeoutMs: 10000,
  });

  for (let i = 0; i < 10; i += 1) {
    const msg = await consumer.receive();
    const pedido = JSON.parse(msg.getData().toString())
    console.log(pedido);
    service.insert(pedido)
    consumer.acknowledge(msg);
  }

  await consumer.close();

})();

File: loja/src/main/java/br/com/saulocn/microservicos/loja/pulsar/PedidosConsumer.java
package br.com.saulocn.microservicos.loja.pulsar;

import org.apache.pulsar.client.api.Consumer;
import org.apache.pulsar.client.api.Message;
import org.apache.pulsar.client.api.PulsarClientException;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.boot.context.event.ApplicationReadyEvent;
import org.springframework.context.event.EventListener;
import org.springframework.stereotype.Component;

@Component
public class PedidosConsumer {
    @Autowired
    PulsarClientSingleton pulsarClientSingleton;

    @EventListener(ApplicationReadyEvent.class)
    public void consume() throws PulsarClientException {
        String subscription = "my-subscription";
        Consumer consumer = pulsarClientSingleton.getClient().newConsumer()
                .topic("pedido_entregue")
                .subscriptionName(subscription)
                .subscribe();
        while (true) {
            Message msg = consumer.receive();
            try {
                System.out.printf("Message received: %s", new String(msg.getData()));
                consumer.acknowledge(msg);
            } catch (Exception e) {
                consumer.negativeAcknowledge(msg);
            }
        }
    }

}
